src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -24,38 +27,97 @@ internal class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
-    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
 
-    static ProcessMetrics()
+    public ProcessMetrics(ProcessInstrumentationOptions options)
     {
+        ThreadSafeInstrumentValues threadSafeInstrumentValues = new();
+
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.usage",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.WorkingSet64;
-            },
+            () => threadSafeInstrumentValues.GetMemoryUsage(),
             unit: "By",
             description: "The amount of physical memory in use.");
 
         // TODO: change to ObservableUpDownCounter
         MeterInstance.CreateObservableGauge(
             "process.memory.virtual",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
-            },
+            () => threadSafeInstrumentValues.GetVirtualMemoryUsage(),
             unit: "By",
             description: "The amount of committed virtual memory.");
     }
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ProcessMetrics"/> class.
-    /// </summary>
-    /// <param name="options">The options to define the metrics.</param>
-    public ProcessMetrics(ProcessInstrumentationOptions options)
+    private class ThreadSafeInstrumentValues
```

Comment:
This layer can be removed as threadID should be universal across threads.

Reply:
Are you saying we could do something like...

```csharp
internal class ProcessMetrics
{
   private static readonly ThreadLocal<InstrumentsValues> currentThreadInstrumentValues = new(() => new InstrumentsValues());
}
```

I think that would work.

Reply:
Thanks for the hint. I also learned why it is a good practice to make ThreadLocal object static!